### ISSUE TITLE
Required lat/lon for sites

### ISSUE THREAD

**Issue by zcrysler** [MotusDev/Motus-TO-DO#111]
Do we have to make lat/lon required fields on the site page?  I'm not able to save any changes to the site page because there's a site with missing lat/lon's but it's a mobile receiver that roams throughout NS and NB: KB-mobile https://motus.org/data/project/sites?projectID=2.  If we want it to remain mandatory I can set it to somewhere in the middle of NS but we aren't requiring a location on the deployments page for mobile receivers

**Comment by Hussell** [comment]
I suppose mobile receiver deployments should not require a site.

**Comment by pmorrill** [comment]
Good point (mobile recvs). We don't know which sites will be associated with which deployments, so I lean towards requiring lat/long on sites, but not requiring sites on mobile recvs. I'll assess that change and let you know.

**Comment by pmorrill** [comment]
I have  posted a change, so that mobile deployments do not generate an error if site is absent.

**Comment by Hussell** [comment]
Catherine, would you remove the sites from all mobile deployments? This would (finally) fix Zoe's problem.

**Comment by denislepage** [comment]
Not sure that's a good idea. Mobile may still be at a local site?




-------- Original message --------

**Comment by pmorrill** [comment]
The system still allows a pi to set a site for a mobile deployment. It just does not require it. Therefore I'd say what we want to do is find any sites that do not include lat / long. and are attached to mobile deployments. Maybe they each need to be looked at?

**Comment by pmorrill** [comment]
I entered this novel in the wrong issue (but have left it there for the moment as it is related):

I believe what we have decided, going forward (next release):

    We will not require lat / long or site for mobile deployments, but we will allow those fields;

    However, if someone tries to set up a new site as part of the deployment creation / edit step, then they MUST provide lat / long as well, for that deployment. (And, I will look at also asking for a landowner)

In the short term, we want to locate all existing sites that are missing lat / long. This should not be allowed! For those sites on the list:

a) if linked only to mobile deployment(s), the site can be de-linked from those deployments and DELETED;

b) if linked to non-mobile deployments (this should not happen!) - we need to deal with them on a case by case basis?
